src/bindings/python/src/compatibility/openvino/inference_engine/ie_api_impl.cpp

```diff
@@ -217,7 +217,7 @@ InferenceEnginePython::IENetwork InferenceEnginePython::read_network(std::string
 InferenceEnginePython::IENetwork::IENetwork(const std::shared_ptr<InferenceEngine::CNNNetwork>& cnn_network)
     : actual(cnn_network) {
     if (actual == nullptr)
-        IE_THROW() << "IENetwork was not initialized.";
+        IE_THROW_G("IENetwork was not initialized.");
```

Comment:
why do you change old macroses? It's API 1.0 and it should be removed before 2024.0 release.
I suppose:
- we need to use new exceptions in all places where it's possible
- only public inference API 1.0 should return old exceptions for compatibility. 
E.g. places like these ones:
https://github.com/openvinotoolkit/openvino/blob/e0ad99081eeca42ffbafd50c473f9835a14669a3/src/inference/src/cpp/ie_infer_request.cpp#L24-L31
Should catch new Exception and rethrow old one.
All other places like IR frontend should directly use new exceptions (and it's a part of migration from 1.0 to 2.0 APIs)

Reply:
The mian reason is to reduce size before API 1.0 exception will be removed.

I apply similar implementation as in API 2.0 exception to wrap throw in exception class function. This approach will produce smaller binary size than overload `<<=` operator an throw. But passing exception message parameters is not compatible with new approach so introduce new macros which throw same exception as API 1.0 but have similar format as OPENVINO_THROW/ASSERT. I think if dropping API 1.0 it will be easier to rework files for new exceptions.

Maybe we should sync with @ilyachur as I could misunderstood concept to rework these macros for size reduction only.